src/util/db/dbconnectionpooled.h

```diff
@@ -0,0 +1,39 @@
+#ifndef MIXXX_DBCONNECTIONPOOLED_H
+#define MIXXX_DBCONNECTIONPOOLED_H
+
+
+#include <QSqlDatabase>
+
+#include "util/db/dbconnectionpool.h"
+#include "util/db/dbconnectionpooler.h"
+
+
+namespace mixxx {
+
+// Dynamically provides thread-local database connections from
+// the pool.
+class DbConnectionPooled final {
+  public:
+    explicit DbConnectionPooled(
+            DbConnectionPoolPtr pDbConnectionPool = DbConnectionPoolPtr())
+        : m_pDbConnectionPool(std::move(pDbConnectionPool)) {
+    }
+    explicit DbConnectionPooled(
+            const DbConnectionPooler& pDbConnectionPooler)
+        : m_pDbConnectionPool(pDbConnectionPooler.m_pDbConnectionPool) {
+    }
+
+    explicit operator bool() const {
```

Comment:
I think this requires a comment. I now understand that checking for m_pDbConnectionPool is sufficiant because "operator QSqlDatabase()" never fails. But it was not easy to discover.